utils/prof_utils.py

```diff
@@ -0,0 +1,1273 @@
+    min_comp_len: float
+        OPTIONAL - Minimum length of a component to be considered real. Measured in bins. If none, will use 1% of total profile lengths + 2. Default: None
+
+    Returns:
+    --------
+    fit_dict: dictionary
+        contains the following keys:
+        W10: float
+            The W10 width of the profile measured in number of bins
+        W10_e: float
+            The uncertainty in the W10
+        W50: float
+            The W50 width of the profile measured in number of bins
+        W50_e: float
+            The uncertainty in the W50
+        Weq: float
+            The equivalent width of the profile measured in number of bins
+        Weq_e: float
+            The uncertainty in the equivalent width
+        Wscat: float
+            The scattering width of the profile measured in number of bins
+        Wscat_e: float
+            The uncertainty in the scattering width
+        maxima: list
+            A lost of floats corresponding to the bin location of each maximum point
+        maxima_e: list
+            A list of floats, each correspinding to the error of the maxima of the same index. Measured in bins
+        redchisq: float
+            The reduced chi sqared of the fit
+        num_gauss: int
+            The number of gaussian components used in the best fit
+        bic: float
+            The Bayesian Information Criterion for the best fit
+        gaussian_params: list
+            A list of length 3*N there N is num_gauss. Each set of 3 parameters corresponds to the amp, centre and width of a guassian component
+        cov_mat: np.matrix
+            The covariance matrix from the fit
+        alpha: float
+            The alpha value used in sigmaClip()
+        profile: list
+            The input profile
+        fit: list
+            The best fit made into a list form
+        sn: float
+            The estimated signal to noise ratio, obtained from the profile. Will be None is period unsupplied
+        sn_e: float
+            The uncertainty in sn. Will be None is period unsupplied
+        scattered: boolean
+            True is the profile is scattered. Will be None is period unsupplied
+    """
+    alphas = np.linspace(1, 5, 9)
+    attempts_dict = {}
+
+    loglvl = logger.level
+    logger.setLevel(logging.WARNING) #squelch logging for the loop
+
+    #loop over the gaussian evaluation fucntion, excepting in-built errors
+    for alpha in alphas:
+        try:
+            prof_dict = prof_eval_gfit(profile, max_N=6, ignore_threshold=ignore_threshold, min_comp_len=min_comp_len, alpha=alpha, period=period)
+            attempts_dict[alpha] = prof_dict
+        except(LittleClipError, LargeClipError, NoComponentsError) as e:
+            logger.setLevel(loglvl)
+            logger.info(e)
+            logger.info("Skipping slpha value: {}".format(alpha))
+            logger.setLevel(logging.WARNING) #squelch logging for the loop
+            pass
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary pass statement](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4980664)